### PR TITLE
MOSIP-32453 : Java 21 migration for sync data & hotlist services

### PR DESCRIPTION
**Pull request by GOKULRAJ136**

### CODE REVIEW

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/config/SwaggerConfig.java [hunk 1]
```diff
@@ -49,11 +48,11 @@ public OpenAPI openApi() {
 		return api;
     }
 	
-	@Bean
+	/*@Bean
```
Comment (LoganathanSekar7627) [inline admin/hotlist-service/src/main/java/io/mosip/hotlist/config/SwaggerConfig.java:51]: Why is this commented? Can you check by uncommenting?

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/config/HotlistConfig.java [hunk 2]
```diff
@@ -102,7 +101,7 @@ private Map<String, Object> additionalProperties() {
 		jpaProperties.put("hibernate.dialect", dialect);
 		jpaProperties.put("hibernate.temp.use_jdbc_metadata_defaults", Boolean.FALSE);
 		jpaProperties.put("hibernate.implicit_naming_strategy", SpringImplicitNamingStrategy.class.getName());
-		jpaProperties.put("hibernate.physical_naming_strategy", SpringPhysicalNamingStrategy.class.getName());
+		jpaProperties.put("hibernate.physical_naming_strategy", CamelCaseToUnderscoresNamingStrategy.class.getName());
```
Comment (LoganathanSekar7627) [inline admin/hotlist-service/src/main/java/io/mosip/hotlist/config/HotlistConfig.java:104]: Is it tested by running and the DB updates working fine locally?

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/event/HotlistEventHandler.java [hunk 3]
```diff
@@ -50,7 +49,7 @@ public class HotlistEventHandler {
 	private String appId;
 
 	/** The publisher. */
-	@Autowired
+	@Mock
```
Comment (LoganathanSekar7627) [inline admin/hotlist-service/src/main/java/io/mosip/hotlist/event/HotlistEventHandler.java:52]: Mock should not be used in the code other than test classes.

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/config/SyncSwaggerConfig.java [hunk 4]
```diff
@@ -50,11 +49,11 @@ public OpenAPI openApi() {
 		return api;
     }
 	
-	@Bean
+	/*@Bean
 	public GroupedOpenApi groupedOpenApi() {
```
Comment (LoganathanSekar7627) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/config/SyncSwaggerConfig.java:53]: Check by uncommenting this if it works.

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/controller/WebsubCallbackController.java [hunk 5]
```diff
@@ -38,7 +35,7 @@ public class WebsubCallbackController {
     private static final String CERTIFICATE_DATA_SHARE_URL = "certChainDatashareUrl";
     private static final String PARTNER_DOMAIN = "partnerDomain";
 
-    @Autowired
+    @Mock
```
Comment (LoganathanSekar7627) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/controller/WebsubCallbackController.java:38]: Don't mock any object in code other than test classes.

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/websub/PartnerCACertEventSubscriber.java [hunk 6]
```diff
@@ -47,11 +46,11 @@ public class PartnerCACertEventSubscriber /*implements ApplicationListener<Appli
     private String hubUrl;
 
     /** The publisher. */
-    @Autowired
+    @Mock
```
Comment (LoganathanSekar7627) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/websub/PartnerCACertEventSubscriber.java:49]: Don't mock in the code.

**Review by LoganathanSekar7627 — CHANGES_REQUESTED** [review]

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/config/SwaggerConfig.java (hunk 1 above)
Comment (GOKULRAJ136) [reply admin/hotlist-service/src/main/java/io/mosip/hotlist/config/SwaggerConfig.java]: Uncommented  and imported the valid imports for GroupedOpenApi, working as expected

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/config/SyncSwaggerConfig.java (hunk 4 above)
Comment (GOKULRAJ136) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/config/SyncSwaggerConfig.java]: Uncommented  and imported the valid imports for GroupedOpenApi, working as expected

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/event/HotlistEventHandler.java (hunk 3 above)
Comment (GOKULRAJ136) [reply admin/hotlist-service/src/main/java/io/mosip/hotlist/event/HotlistEventHandler.java]: changed into Autowired

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/controller/WebsubCallbackController.java (hunk 5 above)
Comment (GOKULRAJ136) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/controller/WebsubCallbackController.java]: changed into Autowired

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/config/HotlistConfig.java (hunk 2 above)
Comment (GOKULRAJ136) [reply admin/hotlist-service/src/main/java/io/mosip/hotlist/config/HotlistConfig.java:104]: tested locally, working fine as expected.

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/websub/PartnerCACertEventSubscriber.java (hunk 6 above)
Comment (GOKULRAJ136) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/websub/PartnerCACertEventSubscriber.java]: changed into Autowired

**Review by LoganathanSekar7627 — APPROVED** [review]

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/config/HotlistConfig.java [hunk 7]
```diff
@@ -53,7 +52,7 @@ public class HotlistConfig {
 	private String dialect;
 
 	/** The publisher. */
-	@Autowired
+	@Mock
```
Comment (LoganathanSekar7627) [inline admin/hotlist-service/src/main/java/io/mosip/hotlist/config/HotlistConfig.java:55]: Revert to @Autowired.

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/exception/HotlistExceptionHandler.java [hunk 8]
```diff
@@ -116,7 +111,7 @@ protected ResponseEntity<ResponseWrapper<HotlistRequestResponseDTO>> handleHotli
 	 * @param request      the request
 	 * @return the response entity
 	 */
-	@Override
+	//@Override
```
Comment (LoganathanSekar7627) [inline admin/hotlist-service/src/main/java/io/mosip/hotlist/exception/HotlistExceptionHandler.java:114]: Can we put the annotation back.

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/websub/PartnerCACertEventSubscriber.java [hunk 9]
```diff
@@ -51,7 +51,7 @@ public class PartnerCACertEventSubscriber /*implements ApplicationListener<Appli
     protected PublisherClient<String, EventModel, HttpHeaders> publisherClient;
 
     /** The subscribe. */
-    @Autowired
+    @Mock
```
Comment (LoganathanSekar7627) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/websub/PartnerCACertEventSubscriber.java:54]: Revert to @Autowired

**Review by LoganathanSekar7627 — CHANGES_REQUESTED** [review]

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/config/HotlistConfig.java (hunk 7 above)
Comment (GOKULRAJ136) [reply admin/hotlist-service/src/main/java/io/mosip/hotlist/config/HotlistConfig.java]: changed into Autowired.

File: admin/hotlist-service/src/main/java/io/mosip/hotlist/exception/HotlistExceptionHandler.java (hunk 8 above)
Comment (GOKULRAJ136) [reply admin/hotlist-service/src/main/java/io/mosip/hotlist/exception/HotlistExceptionHandler.java]: Used Override

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/websub/PartnerCACertEventSubscriber.java (hunk 9 above)
Comment (GOKULRAJ136) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/websub/PartnerCACertEventSubscriber.java]: changed into Autowired.

**Review by LoganathanSekar7627 — APPROVED** [review]

**Review by dhanendra06 — APPROVED** [review]

**Review by ckm007 — APPROVED** [review]

**Pull request merged**